pkg/k8s/endpointsynchronizer/cep.go

```diff
@@ -225,48 +227,72 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
+					// controller, and the localCEP created on startup will be used.
+					if localCEP == nil {
+						localCEP, err = ciliumClient.CiliumEndpoints(namespace).Get(podName, meta_v1.GetOptions{})
+						switch {
+						// The CEP doesn't exist in k8s. This is unexpetected but may occur
+						// if the endpoint was removed from k8s but not yet within the agent.
+						// Mark the CEP for creation on the next controller iteration. This
+						// may never occur if the controller is stopped on Endpoint delete.
+						case err != nil && k8serrors.IsNotFound(err):
+							needInit = true
+							return err
+
+						// We cannot read the upstream CEP. needInit will cause the next
+						// iteration to delete and create the CEP. This is an unexpected
+						// situation.
+						case err != nil && k8serrors.IsInvalid(err):
+							scopedLog.WithError(err).Warn("Invalid CEP during update")
+							needInit = true
+							return nil
+
+						// A real error
+						case err != nil:
+							scopedLog.WithError(err).Error("Cannot get CEP during update")
+							return err
+						}
+					}
 
-					// We cannot read the upstream CEP. needInit will cause the next
-					// iteration to delete and create the CEP. This is an unexpected
-					// situation.
-					case err != nil && k8serrors.IsInvalid(err):
-						scopedLog.WithError(err).Warn("Invalid CEP during update")
-						needInit = true
-						return nil
-
-					// A real error
-					case err != nil:
-						scopedLog.WithError(err).Error("Cannot get CEP during update")
+					// We have an object to reuse. Update and push it up. In the case of an
+					// update error, we retry in the next iteration of the controller using
+					// the copy returned by Update.
+					scopedLog.Debug("Updating CEP from local copy")
+					switch {
+					case ciliumUpdateStatusVerConstr.Check(k8sServerVer):
+						localCEP, err = ciliumClient.CiliumEndpoints(namespace).UpdateStatus(localCEP)
+						mdl.DeepCopyInto(&localCEP.Status)
+						_, err = ciliumClient.CiliumEndpoints(namespace).UpdateStatus(localCEP)
+					default:
+						localCEP, err = ciliumClient.CiliumEndpoints(namespace).Update(localCEP)
+						mdl.DeepCopyInto(&localCEP.Status)
+						_, err = ciliumClient.CiliumEndpoints(namespace).Update(localCEP)
 						return err
 					}
 				}
 
-				// We have an object to reuse. Update and push it up. In the case of an
-				// update error, we retry in the next iteration of the controller using
-				// the copy returned by Update.
-				scopedLog.Debug("Updating CEP from local copy")
-				// Update the copy of the cep
-				mdl.DeepCopyInto(&localCEP.Status)
-				switch {
-				case ciliumUpdateStatusVerConstr.Check(k8sServerVer):
-					localCEP, err = ciliumClient.CiliumEndpoints(namespace).UpdateStatus(localCEP)
-				default:
-					localCEP, err = ciliumClient.CiliumEndpoints(namespace).Update(localCEP)
-				}
-
 				// Handle Update errors or return successfully
 				switch {
```

Comment:
Is this code still reachable? we `return err` from the default switch case of the default further up.

Reply:
removed the return, bad rebase. but yes it was reachable for the PATCH code path